src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/ui/MemoryPanel.java

```diff
@@ -98,23 +104,59 @@ public int getRowCount() {
           return numVisibleRows;
         }
         public int getColumnCount() {
-          return 2;
+            return isAnnotated ? 1 : 2;
         }
         public Object getValueAt(int row, int column) {
-          switch (column) {
-          case 0:  return bigIntToHexString(startVal.add(new BigInteger(Integer.toString((row * addressSize)))));
-          case 1: {
-            try {
-              Address addr = bigIntToAddress(startVal.add(new BigInteger(Integer.toString((row * addressSize)))));
+          // When not annotated, we just display the address followed by its contents in two
+          // separate columns. When annotated the format is just one column which contains:
+          //   <address>: <contents> <PointerFinder output>
+          // For example:
+          //   0x00007f7eb010c330: 0x00007f7eb6c9dfb0 vtable for os::PlatformMonitor + 0x10
+          if (!isAnnotated) {
+            switch (column) {
+            case 0:  return bigIntToHexString(startVal.add(new BigInteger(Integer.toString((row * addressSize)))));
+            case 1: {
+              try {
+                Address addr = bigIntToAddress(startVal.add(new BigInteger(Integer.toString((row * addressSize)))));
+                if (addr != null) {
+                  return addressToString(addr.getAddressAt(0));
+                }
+                return unmappedAddrString;
+              } catch (UnmappedAddressException e) {
+                return unmappedAddrString;
+              }
+            }
+            default: throw new RuntimeException("Column " + column + " out of bounds");
+            }
+          } else {
+            switch (column) {
+            case 0: {
+              String col1;
+              String col2 = unmappedAddrString;
+              String col3 = "";
+              BigInteger bigaddr = startVal.add(new BigInteger(Integer.toString((row * addressSize))));
+              Address addr = bigIntToAddress(bigaddr);
+
+              col1 = bigIntToHexString(bigaddr);
```

Comment:
Nit: This can be simplified just a little bit by moving the lines 137-138 above line 134.
So, it will be like this:
```
              BigInteger bigaddr = startVal.add(new BigInteger(Integer.toString((row * addressSize))));
              Address addr = bigIntToAddress(bigaddr);
              String col1 = bigIntToHexString(bigaddr);
              String col2 = unmappedAddrString;
              String col3 = "";
```

Reply:
ok, I updated it.